tsfc/coffee_mode.py

```diff
@@ -272,7 +271,7 @@ def factorise_atomics(monomial_sum, optimal_atomics, argument_indices):
                 new_atomics.append(new_node)
             else:
                 new_rest = new_node
-        new_monomial_sum.add(sum_indices, new_atomics, new_rest)
+        new_monomial_sum.add(tuple(sum_indices_frozen_set), new_atomics, new_rest)
```

Comment:
Actually I don't need to pass in sum_indices any more as the monomials all have the same sum_indices in this function. I guess I will just pick it from the first term.